ote_sdk/ote_sdk/tests/entities/shapes/test_rectangle.py

```diff
@@ -0,0 +1,672 @@
+        Check Rectangle __hash__ method
+
+        <b>Input data:</b>
+        Instance of Rectangle class
+
+        <b>Expected results:</b>
+        Test passes if __hash__ method returns expected value
+
+        <b>Steps</b>
+        1. Check value returned by __hash__ method
+        """
+        rectangle = self.horizontal_rectangle()
+        assert hash(rectangle) == hash(str(rectangle))
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_rectangle_clip_to_visible_region(self):
+        """
+        <b>Description:</b>
+        Check Rectangle clip_to_visible_region method
+
+        <b>Input data:</b>
+        Rectangle class initiation parameters
+
+        <b>Expected results:</b>
+        Test passes if clip_to_visible_region method return correct value
+
+        <b>Steps</b>
+        1. Check values returned by clip_to_visible_region method for 0<x1<x2, 0<y1<y2, x1<x2<1, y1<y2<1
+        2. Check values returned by clip_to_visible_region method for x1<0, y1<0, x2>1, y2>1
+        3. Check values returned by clip_to_visible_region method for x1=0, y1=0, x2=1, y2=1
+        4. Check ValueError exception raised if x1<0 and x1<x2<0: clipped Rectangle width will be equal 0
+        5. Check ValueError exception raised if 1<x1<x2 and x2>1: clipped Rectangle width will be equal 0
+        6. Check ValueError exception raised if y1<0 and y1<y2<0: clipped Rectangle height will be equal 0
+        7. Check ValueError exception raised if 1<y1<y2 and y2>1: clipped Rectangle height will be equal 0
+        """
+        positive_scenarios = [
+            {
+                "input_params": {
+                    "x1": 0.3,
+                    "y1": 0.2,
+                    "x2": 0.6,
+                    "y2": 0.4,
+                    "labels": self.rectangle_labels(),
+                },
+                "params_expected": {"x1": 0.3, "y1": 0.2, "x2": 0.6, "y2": 0.4},
+            },
+            {
+                "input_params": {
+                    "x1": -0.2,
+                    "y1": -0.3,
+                    "x2": 1.6,
+                    "y2": 1.4,
+                    "labels": self.rectangle_labels(),
+                },
+                "params_expected": {"x1": 0.0, "y1": 0.0, "x2": 1.0, "y2": 1.0},
+            },
+            {
+                "input_params": {
+                    "x1": 0.0,
+                    "y1": 0.0,
+                    "x2": 1.0,
+                    "y2": 1.0,
+                    "labels": self.rectangle_labels(),
+                },
+                "params_expected": {"x1": 0.0, "y1": 0.0, "x2": 1.0, "y2": 1.0},
+            },
+        ]
+        for scenario in positive_scenarios:
+            rectangle_actual = Rectangle(**scenario.get("input_params"))
+            rectangle_expected = Rectangle(**scenario.get("params_expected"))
+            rectangle_actual.modification_date = self.modification_date
```

Comment:
the same issue

Reply:
modification_date attribute belongs to class TestRectangle and is evaluated once, should I rename it?

Reply:
sorry, I what thinking that there is `assert rectangle_actual.modification_date == self.modification_date `
Misread.
Nevermind.